recorder/recorder.go

```diff
@@ -157,6 +178,14 @@ type Options struct {
 	// the real requests
 	RealTransport http.RoundTripper
 
+	// Block unsafe methods from ever being called with RealTransport.
+	// The definition of "Safe Methods" comes from
+	// https://datatracker.ietf.org/doc/html/rfc2616#section-9.1.1
```

Comment:
I think we should actually refer to [RFC 9110, section 9.2.1](https://datatracker.ietf.org/doc/html/rfc9110#name-safe-methods) here instead, since it clearly specifies that `GET`, `HEAD`, `OPTIONS` and `TRACE` are safe methods.